homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -88,11 +89,13 @@ async def async_step_reauth(self, data: dict[str, str]) -> FlowResult:
         return await self.async_step_link()
 
     async def async_step_zeroconf(
-        self, discovery_info: DiscoveryInfoType
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
     ) -> FlowResult:
         """Handle Nanoleaf Zeroconf discovery."""
         _LOGGER.debug("Zeroconf discovered: %s", discovery_info)
-        return await self._async_homekit_zeroconf_discovery_handler(discovery_info)
+        return await self._async_homekit_zeroconf_discovery_handler(
+            cast(dict, discovery_info)
```

Comment:
Once homekit is also migrated to use ZeroconfServiceInfo (or sub class) we can remove the cast.